### PR TITLE
[Create new community popup]: Adjusted request to join

### LINKED ISSUE #12809: Create new / edit community flow: Warning text on "Request to join required" option
A small tweak has been made to the `Request to join required` checkbox list item featured in the `Create New Community` and `Edit Community` flows.

Changes made:

1) `Request to join required` option moved to base of checkbox list
2) When the checkbox for this item is selected, a warning in orange text is shown reading:
`Warning: Only token gated communities (or token gated channels inside non-token gated community) are encrypted` 

NB this text is only shown when the checkbox is in the selected state. The rest of the UI should push down as required to accommodate this text as the checkbox is selected.

**Change to Create New Community flow:**
https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=2636-359221&mode=design&t=3AguBBoHdwrIIKGc-4

Screenshot 2023-11-21 at 11 17 26 AM

**Change to Edit Community flow:**
https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=45510-224401&mode=design&t=3AguBBoHdwrIIKGc-4

Original:
Screenshot 2023-11-21 at 11 18 21 AM

New:
Screenshot 2023-11-21 at 11 19 06 AM

cc @John-44

### PR DESCRIPTION
**Pull request by alexandraB99**
As per new designs

Closes #12809

### What does the PR do
Create new community popup: Adjusted request to join checkbox

### Affected areas
Create new community popup

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/31625338/7be5019f-f596-41ef-b475-05aab49f9ffa

### CODE REVIEW

File: ui/app/AppLayouts/Communities/controls/Options.qml [hunk 1]
```diff
@@ -49,33 +49,51 @@ Column {
 
         StatusBaseText {
             Layout.fillWidth: true
-            text: qsTr("Request to join required")
+            text: qsTr("Any member can pin a message")
             font.pixelSize: Theme.primaryTextFontSize
             TapHandler {
-                onTapped: requestToJoinToggle.toggle()
+                onTapped: pinMessagesToggle.toggle()
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/controls/Options.qml:55]: Is this connected to some backend thing?

File: ui/app/AppLayouts/Communities/controls/Options.qml [hunk 2]
```diff
@@ -49,33 +49,51 @@ Column {
...
 
-    RowLayout {
+    ColumnLayout {
         width: parent.width
-        height: d.optionHeight
+        height: d.optionHeight + (requestToJoinToggle.checked ? (warningText.height) : 0)
+        spacing: 0
 
-        StatusBaseText {
+        RowLayout {
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/controls/Options.qml:69]: You can use just the `StatusCheckBox` with:
```qml
LayoutMirroring.enabled: true
LayoutMirroring.childrenInherit: true
```

Comment (caybro) [reply ui/app/AppLayouts/Communities/controls/Options.qml:69]: (to flip the position of the text)

Comment (caybro) [reply ui/app/AppLayouts/Communities/controls/Options.qml:69]: or... it already has this property
```qml
property bool leftSide: true
```

File: ui/app/AppLayouts/Communities/controls/Options.qml [hunk 3]
```diff
@@ -49,33 +49,51 @@ Column {
...
+                id: requestToJoinToggle
             }
         }
 
-        StatusCheckBox {
-            id: pinMessagesToggle
+        StatusBaseText {
+            id: warningText
+            Layout.fillWidth: true
+            Layout.preferredHeight: visible ? contentHeight : 0
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/controls/Options.qml:90]: Not needed imo

File: ui/app/AppLayouts/Communities/controls/Options.qml [hunk 4]
```diff
@@ -49,33 +49,51 @@ Column {
...
-            id: requestToJoinToggle
+            id: pinMessagesToggle
         }
     }
 
-    RowLayout {
+    ColumnLayout {
         width: parent.width
-        height: d.optionHeight
+        height: d.optionHeight + (requestToJoinToggle.checked ? (warningText.height) : 0)
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/controls/Options.qml:66]: This shouldn't be necessary; the height will adjust based on what elements are visible

Comment (caybro) [reply ui/app/AppLayouts/Communities/controls/Options.qml:58]: Really needed?

File: ui/app/AppLayouts/Communities/controls/Options.qml (hunk 3 above)
Comment (caybro) [reply ui/app/AppLayouts/Communities/controls/Options.qml:75]: Same here

Comment (alexandraB99) [reply ui/app/AppLayouts/Communities/controls/Options.qml]: done :)

**Review by caybro — APPROVED** [review]
LGTM

**Review by alexjba — APPROVED** [review]
LGTM!

**Pull request merged**